packages/flutter/lib/src/gestures/events.dart

```diff
@@ -908,7 +908,7 @@ class PointerEnterEvent extends PointerEvent {
     down: event?.down,
     synthesized: event?.synthesized,
     transform: event?.transform,
-    original: event?.original as PointerEnterEvent,
+    original: null,
```

Comment:
An error is thrown in this as-cast when converting other event types using `fromMouseEvent` (e.g. from `PointerDownEvent` to `PointerEnterEvent`). This as-cast was not here when this constructor was first created, but is reasonable since the constructor requires `original` to be `PointerEnterEvent`, which also makes sense since `original` was added for the transformed constructor to **transform between coordinate space**. So if we think it this way, this `original` should be null because `original` was not defined to suit in the type converting case.